File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Aruba.m
//
//  UIImage+Aruba.m
//  CountriesFlags
//

#import "Countries/UIImage+Aruba.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Aruba)

+ (UIImage *)flagArubaWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:65.0f/255.0f
                                         green:143.0f/255.0f
                                          blue:222.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:1.0f
                                           green:210.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:246.0f/255.0f
                                        green:52.0f/255.0f
                                         blue:63.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 18.0f * 12.0f, size.width, size.height / 18.0f));
    UIRectFill(CGRectMake(0.0f, size.height / 18.0f * 14.0f, size.width, size.height / 18.0f));
    
    [colorRed setFill];
    [colorWhite setStroke];
    UIBezierPath* path = [UIBezierPath starAtPoint:CGPointMake(size.width / 27.0f * 4.0f, size.height / 18.0f * 4.0f)
                                            radius:size.height / 18.0f * 2.68f
                                              type:FlagStarTypeFourPointed];
    path.lineWidth = size.height / 95.0f;
    [path fill];
    [path stroke];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Aruba.swift
//
//  UIImage+Aruba.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Aruba "wikipedia"
     
     Generates the flag of Aruba.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Aruba.
     */
    internal static func flagAruba(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 65.0 / 255.0,
                                green: 143.0 / 255.0,
                                blue: 222.0 / 255.0,
                                alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 210.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)
        let colorRed = UIColor(red: 246.0 / 255.0,
                               green: 52.0 / 255.0,
                               blue: 63.0 / 255.0,
                               alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorYellow.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 18.0 * 12.0),
                          size: CGSize(width: size.width, height: size.height / 18.0)))
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 18.0 * 14.0),
                          size: CGSize(width: size.width, height: size.height / 18.0)))

        colorWhite.setFill()
        UIBezierPath.star(at: CGPoint(x: size.width / 27.0 * 4.0, y: size.height / 18.0 * 4.0),
                          radius: size.height / 18.0 * 3.0,
                          type: .fourPointed).fill()

        colorRed.setFill()
        UIBezierPath.star(at: CGPoint(x: size.width / 27.0 * 4.0, y: size.height / 18.0 * 4.0),
                          radius: size.height / 18.0 * 2.3,
                          type: .fourPointed).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
